Fix #2004, pronoun tagging to new_cat_welcoming.json

tailhaver opened the pull request:
Fixes #2004 by manually appending "kit" or "paw" to loners if they are under a certain age. This fixes an edge case where a kit or app is a loner, giving them a proper name instead of a warrior name. This is implemented _after_ the standard suffix check for kits/apps, just as a fallback.

This also adds pronoun tagging to [new_cat_welcoming](https://github.com/ClanGenOfficial/clangen/blob/development/resources/dicts/patrols/new_cat_welcoming.json), my first pronoun tags! I'd prefer if someone looked over it just as a double check, although worst case someone will find it and (hopefully) report it.

Commit:
fix #2004 hopefully

Commit:
pronoun tagging, + fix gender on patrol listed in #2004

Commit:
prevent non-loner cats under 12 moons from falsely getting assigned as kits, apprentices

Commit:
fix incorrectly tagged patrol outcomes

Tiritirimatangi reviewed (commented):
Tiri has reviewed the typo and pronoun tagging fixes and would like to fall at your feet in gratitude - I'm not qualified to review the coding stuff though

Thlumyn commented on resources/dicts/patrols/new_cat_welcoming.json:
```diff
@@ -150,7 +150,7 @@
         "antag_success_outcomes": [
             {
-                "text": "p_l approaches the loner and tells them firmly that newcomers aren't being taken in right now. The loner seems disappointed, but leaves anyway.",
+                "text": "p_l approaches the loner and tells themfirmly that newcomers aren't being taken in right now. The loner seems disappointed, but leaves anyway.",
```
Spelling (themfirmly)

Thlumyn commented on scripts/cat/names.py:
```diff
@@ -173,12 +175,16 @@ def give_suffix(self, pelt, biome, tortiepattern):
-            if game.config['fun']['april_fools']:
-                return self.prefix + 'egg'
-            return self.prefix + self.suffix
+        if self.status in ['loner'] and self.moons is not None and self.moons < 12:
```
Can we add `and not self.specsuffix_hidden` here? So that a loner with no suffix doesn't get paw or kit attached

Thlumyn commented on scripts/cat/names.py:
```diff
@@ -173,12 +175,16 @@ def give_suffix(self, pelt, biome, tortiepattern):
-            if game.config['fun']['april_fools']:
-                return self.prefix + 'egg'
-            return self.prefix + self.suffix
+        if self.status in ['loner'] and self.moons is not None and self.moons < 12:
```
Moons <12 doesn't necessarily line up with being an apprentice; a lot of cases here where that wouldn't line up. Such as when players have exp based graduation turned on and the cats graduate based on exp. If you really wanted to, you could check for exp based graduation and check the cat's exp to see if they are really meant to be an apprentice. But that leaves a bit of code redundancy - if anything changes with the app/exp code it could mess up here. Maybe the apprentice graduation already has a function you could borrow for that. 
Though that might not fix things in the case of, like, Ravenpaw, where he kept his apprentice name his whole life after leaving. Not sure there's a good way to approach this.

tailhaver replied:
This will only be used for loners outside the clan though, such as in one of the outcomes for `gen_bord_welcomingsparrowgift1` where you meet a loner kitten. It would appear as if loners don't gain any experience, and if they join the clan then they have their loner status stripped.
I personally don't see another way to fix this outside of potentially adding an `is_loner` flag to each cat, but that sounds unneeded.

Commit:
Fix crash, fix 2/3 requested changes
The game would crash upon timeskip when you create a new classic game while having an expanded mode game open.
